summingbird-scalding-test/src/test/scala/com/twitter/summingbird/scalding/BatchedStoreProperties.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ Copyright 2013 Twitter, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package com.twitter.summingbird.scalding
+
+import cascading.flow.{ Flow, FlowDef }
+
+import com.twitter.algebird._
+import com.twitter.algebird.monad._
+import com.twitter.summingbird.batch._
+import com.twitter.scalding.{ Source => ScaldingSource, Test => TestMode, _ }
+
+import org.scalacheck._
+import org.scalacheck.Prop._
+import org.scalacheck.Properties
+
+/**
+ * Tests for Summingbird's Scalding planner.
+ */
+
+object BatchedStoreProperties extends Properties("BatchedStore's Properties") {
+
+  implicit def intervalArb[T: Arbitrary: Ordering]: Arbitrary[Interval[T]] =
+    Arbitrary {
+      for {
+        l <- Arbitrary.arbitrary[T]
+        u <- Arbitrary.arbitrary[T]
+      } yield Interval.leftClosedRightOpen(l, u)
+    }
+
+  implicit val arbTimestamp: Arbitrary[Timestamp] = Arbitrary {
+    // a relevant 200 or so year range
+    Gen.choose(1L, 100000L)
+      .map { Timestamp(_) }
+  }
+
+  def sample[T: Arbitrary]: T = Arbitrary.arbitrary[T].sample.get
+
+  implicit def arbitraryPipeFactory: Arbitrary[PipeFactory[(Int, Int)]] = {
+    Arbitrary {
+      Gen.const {
+        StateWithError[(Interval[Timestamp], Mode), List[FailureReason], FlowToPipe[(Int, Int)]] {
+          (timeMode: (Interval[Timestamp], Mode)) =>
+            {
+              val (time: Interval[Timestamp], mode: Mode) = timeMode
+              val a: FlowToPipe[(Int, Int)] = Reader { (fdM: (FlowDef, Mode)) => TypedPipe.from[(Timestamp, (Int, Int))](Seq((Timestamp(10), (2, 3)))) }
+              Right((timeMode, a))
+            }
+        }
+      }
+    }
+  }
+
+  implicit def timeExtractor[T <: (Long, _)] = TestUtil.simpleTimeExtractor[T]
+
+  implicit def arbitraryInputWithTimeStampAndBatcher: Arbitrary[(List[(Long, Int)], Batcher, TestStore[Int, Int])] = Arbitrary {
```

Comment:
I think it is better to use `Gen[T]` here for such a particular type as this, and then explicitly pass the Gen rather than Arbitrary.